### PR TITLE
linter and test_aider_linter extensions for eslint

### PR DESCRIPTION
**Pull request by tobitege**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Followup to #3452 `(feat) implement typescript linting for CodeActAgent`
Added methods trying to lint with `eslint` and added tests to `test_aider_linter.py`.

I've put this branch up on OpenHands, so maintainers have full access.

@amanape @neubig Would love to get your feedback, whether the way I'm trying to use
the installed eslint modules from the `frontend` folder makes sense or should be different
or how the tests are set up etc.?

---
**Other references**

#3431 `[Bug]: Underspecified linter errors (in JavaScript?)`

### CODE REVIEW

**Comment by tobitege** [comment]
Ughh... all ran fine in WSL, but not CI... back to drawing board...

File: openhands/runtime/plugins/agent_skills/utils/aider/linter.py [hunk 1]
```diff
@@ -131,7 +139,100 @@ def check_ts_installed(self):
...
+            },
+            'settings': {'react': {'version': 'detect'}},
+        }
+
+        # Write config to a temporary file
+        with tempfile.NamedTemporaryFile(
+            mode='w', suffix='.json', delete=False
+        ) as temp_config:
+            json.dump(eslint_config, temp_config)
+            temp_config_path = temp_config.name
```
Comment (neubig) [inline openhands/runtime/plugins/agent_skills/utils/aider/linter.py:187]: I'm a bit confused by this with syntax. Why not just use a normal with statement and put the remainder of the function inside the with statement? Then I think you wouldn't need all the `os.unlink`s

File: openhands/runtime/plugins/agent_skills/utils/aider/linter.py [hunk 2]
```diff
@@ -131,7 +139,100 @@ def check_ts_installed(self):
...
+
+    def print_lint_result(self, lint_result):
+        if not isinstance(lint_result, LintResult):
+            return
+        print(f'\n{lint_result.text.strip()}')
+        if isinstance(lint_result.lines, list) and lint_result.lines:
+            if isinstance(lint_result.lines[0], LintResult):
+                self.print_lint_result(lint_result.lines[0])
+
+    def ts_eslint(self, fname, rel_fname, code):
```
Comment (neubig) [inline openhands/runtime/plugins/agent_skills/utils/aider/linter.py:158]: Could you please add type annotations?

File: openhands/runtime/plugins/agent_skills/utils/aider/linter.py [hunk 3]
```diff
@@ -131,7 +139,100 @@ def check_ts_installed(self):
...
-    def ts_lint(self, fname, rel_fname, code):
+    def check_ts_installed(self) -> bool:
+        """Check if TypeScript is installed."""
+        return self._check_tool_installed('tsc')
+
+    def check_eslint_installed(self) -> bool:
+        """Check if ESLint is installed."""
+        return self._check_tool_installed('eslint')
+
+    def print_lint_result(self, lint_result):
```
Comment (neubig) [inline openhands/runtime/plugins/agent_skills/utils/aider/linter.py:150]: Could you please add type annotations?

**Review by neubig — COMMENTED** [review]
Thanks for this! I had a few suggestions.

Also, as an auxiliary note, I think we should probably rename this away from `aider_linter` now, since:
1. we've added far above and beyond the aider linter
2. if we rename the directory just `linter` then we could have `linter/base.py`, `linter/python.py`, `linter/javascript.py` to make the files a little bit less overwhelming

We can do that in a separate PR though!

**Comment by tobitege** [comment]
> Thanks for this! I had a few suggestions.
> 
> Also, as an auxiliary note, I think we should probably rename this away from `aider_linter` now, since:
> 
>     1. we've added far above and beyond the aider linter
> 
>     2. if we rename the directory just `linter` then we could have `linter/base.py`, `linter/python.py`, `linter/javascript.py` to make the files a little bit less overwhelming
> 
> 
> We can do that in a separate PR though!

Thanks for the feedback! Except the file splitup, I've resolved the mentioned issues.
Re: the one `with` statement I had to go with a try/finally block, the other way did not work as expected.

**Review by neubig — APPROVED** [review]
I think in absence of evidence to the contrary this seems to be a net positive, so I'm OK with approving now!

**Comment by tobitege** [comment]
> I think in absence of evidence to the contrary this seems to be a net positive, so I'm OK with approving now!

Thank you! 👍

**Pull request merged**